agent/engine/docker_task_engine_test.go

```diff
@@ -1660,6 +1660,42 @@ func TestUpdateContainerReference(t *testing.T) {
 	assert.True(t, imageState.PullSucceeded, "PullSucceeded set to false")
 }
 
+// TestPullAndUpdateContainerReference checks whether a container is added to task engine state when
+// PullSucceeded of the image is fulfilled.
+func TestPullAndUpdateContainerReference(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+	ctrl, client, _, privateTaskEngine, _, imageManager, _ := mocks(t, ctx, &config.Config{})
+	defer ctrl.Finish()
+
+	taskEngine, _ := privateTaskEngine.(*DockerTaskEngine)
+	taskEngine._time = nil
+	imageName := "image"
+	container := &apicontainer.Container{
+		Type:  apicontainer.ContainerNormal,
+		Image: imageName,
+	}
+	task := &apitask.Task{
+		Containers: []*apicontainer.Container{container},
+	}
+	imageState := &image.ImageState{
+		Image: &image.Image{ImageID: "id"},
+	}
+	taskEngine.State().AddContainer(&apicontainer.DockerContainer{
```

Comment:
why are we adding to the state here? shouldn't the code be doing this after image pull succeeds?

Reply:
The state is removed in [this commit](https://github.com/aws/amazon-ecs-agent/pull/2700/commits/fba2d68cb35ed883fb1b920417c02caa77a43981#diff-aa8aba86b2dbb7c182fa55d44aed378f40e350d5ae11405e017dd56d6b58ad04R1689). Thank you!